src/components/cards/LearnCard.tsx

```diff
@@ -0,0 +1,122 @@
+import { Box, Column, Columns, Inline, Stack, Text } from '@/design-system';
+import { useTheme } from '@/theme';
+import React, { useCallback, useReducer } from 'react';
+import { GenericCard } from './GenericCard';
+import { LearnCardDetails, learnCards, learnCategoryColors } from './constants';
+import { IconOrb } from './reusables/IconOrb';
+import { useNavigation } from '@/navigation';
+import Routes from '@/navigation/routesNames';
+import { delay } from '@/helpers/utilities';
+import lang from 'i18n-js';
+
+export const LearnCardHeight = 184;
+
+interface LearnCardProps {
+  cardDetails?: LearnCardDetails;
+  type: 'square' | 'stretch';
+}
+
+export const LearnCard = ({ cardDetails, type }: LearnCardProps) => {
+  const { navigate } = useNavigation();
+  const { isDarkMode } = useTheme();
+  const [index, incrementIndex] = useReducer(
```

Comment:
all this logic should be moved outside of `LearnCard` to a separate component the renders` LearnCards`, called it `LearnSlides` or smthn

Reply:
what's the advantage of this? the index state is the only logic that is unique to the dynamic learn card so i don't think it's worth creating a whole other component just to extract that